File: Dec2020/12Dec/BishuAndSoldiers.java
/*
Problem Name: Bishu and Soldiers
Problem Link: https://www.hackerearth.com/practice/algorithms/searching/binary-search/practice-problems/algorithm/bishu-and-soldiers/
*/
import java.io.*;
import java.util.*;
class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner in = new Scanner(System.in);
        PrintWriter out = new PrintWriter(System.out);
        int n = in.nextInt();
        int power[] = new int[n];
        for(int i=0;i<n;i++)
            power[i] = in.nextInt();
        Arrays.sort(power);
        int q = in.nextInt();
        for(;q>0;q--){
            int m = in.nextInt();
            int ind = bs(power,m);
            out.print((ind+1)+" ");
            int sum = 0;
            for(int i=0;i<=ind;i++)
                sum += power[i];
            out.println(sum);
        }
        out.flush();
    }
    static int bs(int arr[], int k){
        int l = 0, r = arr.length - 1, mid = 0;
        for(;l<=r;){
            mid = (l+r)/2;
            if(arr[mid] <= k)
                l = mid + 1;
            else
                r = mid - 1; 
        }
        return r;
    }
}

File: Dec2020/12Dec/BishuAndSoldiers.cpp
/*
Problem Name: Bishu and Soldiers
Problem Link: https://www.hackerearth.com/practice/algorithms/searching/binary-search/practice-problems/algorithm/bishu-and-soldiers/
*/

#include<bits/stdc++.h>
using namespace std;

#define test(t)     int t; cin>>t; while(t--)
#define ll          long long

int main(){
    ios::sync_with_stdio(0);    
    cin.tie(0);             
    ll n, q;
    cin>>n;
    ll soldiers[n];
    for(ll i=0;i<n;++i)
        cin>>soldiers[i];
    cin>>q;
    ll bishu[q];
    for(ll i=0;i<q;++i)
        cin>>bishu[i];
    
    ll sum, num;
    sort(soldiers,soldiers+n);
    for(int i=0;i<q;++i){
        sum = 0, num = 0;
        for(int j=0;j<n;++j){
            if(soldiers[j]<=bishu[i]){
                sum+=soldiers[j];
                num++;
            }
            else
                break;
        }
        cout<<num<<" "<<sum<<"\n";
    }
    return 0;
}
